examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx

```diff
@@ -0,0 +1,172 @@
+
+  const map = useGoogleMap();
+  const autocompleteService = useAutocompleteService();
+  const placesService = usePlacesService();
+
+  const maxNumberOfSuggestions = 5;
+
+  // Update the user input value
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+
+    if (timeout.current) {
+      clearTimeout(timeout.current);
+    }
+
+    // Update debounced value (for search request) after delay
+    timeout.current = setTimeout(() => {
+      setDebouncedValue(event.target.value);
+
+      // Show dropdown
+      if (!suggestionsAreVisible) {
+        setSuggestionsAreVisible(true);
+      }
+    }, 300);
+  };
+
+  // Handle suggestion selection
+  const selectSuggestion = (
+    suggestion: PlacesAutocompleteServiceSuggestion
+  ) => {
+    inputRef.current?.focus();
+    setInputValue(suggestion.label);
+
+    // Close dropdown
+    setSuggestionsAreVisible(false);
+
+    // Get the location from Places Service of the selected place and zoom to it
+    placesService?.getDetails(
+      {placeId: suggestion.id},
+      (
+        placeResult: google.maps.places.PlaceResult | null,
+        status: google.maps.places.PlacesServiceStatus
+      ) => {
+        if (
+          status !== google.maps.places.PlacesServiceStatus.OK ||
+          !placeResult
+        ) {
+          return;
+        }
+
+        // Get position of the suggestion to move map
+        const position = placeResult.geometry?.location;
+
+        if (map && position) {
+          map.setZoom(14);
+          map.panTo(position);
+        }
+      }
+    );
+  };
+
+  // Update suggestions and get autocomplete place suggestions
+  useEffect(() => {
+    if (inputValue.length >= 2) {
+      autocompleteService?.getPlacePredictions(
+        {
+          input: inputValue
+        },
+        (
+          predictions: google.maps.places.AutocompletePrediction[] | null,
+          status: google.maps.places.PlacesServiceStatus
+        ) => {
+          if (
+            status !== google.maps.places.PlacesServiceStatus.OK ||
+            !predictions
+          ) {
+            return;
+          }
+
+          const autocompleteSuggestions = predictions
+            .slice(0, maxNumberOfSuggestions)
+            .map(prediction => ({
+              id: prediction.place_id,
+              label: prediction.description
+            }));
+
+          // Update suggestions for dropdown suggestions list
+          setSuggestions(autocompleteSuggestions);
+        }
+      );
+    } else {
+      setSuggestions([]);
+    }
+  }, [inputValue]);
+
+  return (
+    <>
+      <input
+        ref={inputRef}
+        className={cx(
+          styles.searchInput,
+          debouncedInputValue.length && styles.expanded
+        )}
+        value={inputValue}
+        onChange={handleInputChange}
+        autoComplete="off"
+        aria-owns="search-suggestions"
+        aria-expanded={debouncedInputValue.length >= 3}
```

Comment:
Why `>= 3` here? Line 137 and 152 say it's expanded if `debouncedInputValue.length`. Shouldn't it be the same condition in all places?

Reply:
Actually, I am not using the styles anymore, so I can just remove it :) But good you noticed that here!